dll/win32/advapi32/service/sctrl.c

```diff
@@ -517,31 +517,33 @@ static DWORD
 ScControlService(PACTIVE_SERVICE lpService,
                  PSCM_CONTROL_PACKET ControlPacket)
 {
-    DWORD dwError;
+    DWORD dwError = ERROR_SUCCESS;
 
     TRACE("ScControlService(%p %p)\n",
           lpService, ControlPacket);
 
     if (lpService == NULL || ControlPacket == NULL)
-        return ERROR_INVALID_PARAMETER;
+    {
+        dwError = ERROR_INVALID_PARAMETER;
+        goto done;
+    }
 
     TRACE("Size: %lu\n", ControlPacket->dwSize);
     TRACE("Service: %S\n", (PWSTR)((ULONG_PTR)ControlPacket + ControlPacket->dwServiceNameOffset));
 
     if (lpService->HandlerFunction)
     {
         (lpService->HandlerFunction)(ControlPacket->dwControl);
-        dwError = ERROR_SUCCESS;
     }
     else if (lpService->HandlerFunctionEx)
     {
         /* FIXME: Send correct 2nd and 3rd parameters */
         (lpService->HandlerFunctionEx)(ControlPacket->dwControl,
                                        0, NULL,
                                        lpService->HandlerContext);
-        dwError = ERROR_SUCCESS;
     }
 
```

Comment:
Does clang stop complaining if you just do `else ASSERT(FALSE);` or some variation thereof? That seems simpler.

Reply:
> `else ASSERT(FALSE);`

As expected, that doesn't help.